### PR TITLE
IGNITE-23353 Improve CREATE TABLE syntax

### PR DESCRIPTION
**Pull request by ygerzhedovich**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateTable.java [hunk 1]
```diff
@@ -49,33 +49,37 @@ protected Operator(boolean existFlag) {
         public SqlCall createCall(@Nullable SqlLiteral functionQualifier, SqlParserPos pos,
                 @Nullable SqlNode... operands) {
             return new IgniteSqlCreateTable(pos, existFlag(), (SqlIdentifier) operands[0], (SqlNodeList) operands[1],
-                    (SqlNodeList) operands[2], (SqlNodeList) operands[3]);
+                    (SqlNodeList) operands[2], (SqlIdentifier) operands[3], (SqlLiteral) operands[4]);
         }
     }
 
     private final SqlIdentifier name;
+    @Nullable
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateTable.java:57]: annotation seems to be misplaced

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverterTest.java [hunk 2]
```diff
@@ -613,12 +609,12 @@ public Stream<DynamicTest> timestampWithTzWithDefaults() {
 
     @Test
     public void tableWithAutogenPkColumn() throws SqlParseException {
-        var node = parse("CREATE TABLE t (id uuid default rand_uuid primary key, val int) WITH STORAGE_PROFILE='"
+        var node = parse("CREATE TABLE t (id uuid default rand_uuid primary key, val int) STORAGE PROFILE '"
```
Comment (korlov42) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverterTest.java:612]: `var` is not allowed in this context

File: docs/_docs/sql-reference/ddl.adoc [hunk 3]
```diff
@@ -97,18 +94,7 @@ CREATE TABLE IF NOT EXISTS Person (
...
-Creates a Person table where the records expire at timestamps in the `ttl` column:
-
-[source,sql]
-----
-CREATE TABLE IF NOT EXISTS Person ( 
-  id int PRIMARY KEY,
-  name varchar,
-  ttl timestamp
-) EXPIRE AT ttl
+) ZONE=`MYZONE`
```
[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply docs/_docs/sql-reference/ddl.adoc:97]: agree, the the current state of documentation is bad requiring total rework. But in the patch I just removed use old syntax.

[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply docs/_docs/sql-reference/ddl.adoc:97]: sign `=` will be removed anyway )

File: docs/_docs/developers-guide/java-to-tables.adoc [hunk 4]
```diff
@@ -82,7 +82,7 @@ The result is equivalent to the following SQL multi-statement:
 
 [source, sql]
 ----
-CREATE ZONE IF NOT EXISTS zone_test ENGINE ROCKSDB WITH PARTITIONS=2;
+CREATE ZONE IF NOT EXISTS ZONE_TEST ENGINE ROCKSDB WITH PARTITIONS=2;
```
[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply docs/_docs/developers-guide/java-to-tables.adoc:85]: ok

File: modules/catalog/src/integrationTest/java/org/apache/ignite/internal/catalog/it/ItConcurrentDdlsTest.java [hunk 5]
```diff
@@ -62,7 +62,7 @@ private void createTablesInParallel() {
             String tableName = "TEST" + n;
 
             node(0).sql().executeScript(
-                    "CREATE TABLE " + tableName + " (id INT PRIMARY KEY, val VARCHAR) WITH primary_zone='" + ZONE_NAME + "'"
+                    "CREATE TABLE " + tableName + " (id INT PRIMARY KEY, val VARCHAR) zone " + ZONE_NAME
```
[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply modules/catalog/src/integrationTest/java/org/apache/ignite/internal/catalog/it/ItConcurrentDdlsTest.java:65]: fixed

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractMultiNodeBenchmark.java [hunk 6]
```diff
@@ -132,7 +132,7 @@ protected static void createTable(String tableName, List<String> columns, List<S
             createTableStatement += "\nCOLOCATE BY (" + String.join(", ", colocationKeys) + ")";
         }
 
-        createTableStatement += "\nWITH primary_zone='" + ZONE_NAME + "'";
+        createTableStatement += "\nzone " + ZONE_NAME;
```
[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractMultiNodeBenchmark.java:135]: fixed

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/SqlIndexScanBenchmark.java [hunk 7]
```diff
@@ -92,7 +92,7 @@ public void setUp() throws Exception {
             if (!Files.exists(workDir().resolve(DATASET_READY_MARK_FILE_NAME))) {
                 sql.executeScript(
                         "CREATE ZONE single_partition_zone WITH replicas = 1, partitions = 1;"
-                                + "CREATE TABLE test (id INT PRIMARY KEY, val DATE) WITH primary_zone = single_partition_zone;"
+                                + "CREATE TABLE test (id INT PRIMARY KEY, val DATE) zone single_partition_zone;"
```
[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/SqlIndexScanBenchmark.java:95]: fixed

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java [hunk 8]
```diff
@@ -1758,7 +1757,7 @@ public void tableRecoveryOnMultipleRestartingNodes(int nodeThatWrittenAssignment
         nodeInhibitor2.startInhibit();
 
         sql.executeAsync(null, "CREATE TABLE " + tableName
-                + "(id INT PRIMARY KEY, name VARCHAR) WITH PRIMARY_ZONE='" + zoneName + "';");
+                + "(id INT PRIMARY KEY, name VARCHAR) ZONE" + zoneName + ";");
```
[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java:1760]: fixed

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/schemasync/ItSchemaSyncMultiNodeTest.java [hunk 9]
```diff
@@ -186,7 +186,7 @@ void ddlAfterDdlOnAnotherNodeSeesFirstDdlResults() {
             });
 
             CompletableFuture<Void> tableCreationFuture = runAsync(() -> cluster.doInSession(NODE_1_INDEX, session -> {
-                executeUpdate("CREATE TABLE " + TABLE_NAME + " (id int PRIMARY KEY, val varchar) WITH primary_zone='TEST_ZONE'", session);
+                executeUpdate("CREATE TABLE " + TABLE_NAME + " (id int PRIMARY KEY, val varchar) zone TEST_ZONE", session);
```
[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/schemasync/ItSchemaSyncMultiNodeTest.java:189]: fixed

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverterTest.java (hunk 2 above)
Comment (ygerzhedovich) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverterTest.java:612]: fixed

**Review by lowka — APPROVED** [review]

File: docs/_docs/sql-reference/grammar-reference.adoc [hunk 10]
```diff
@@ -180,38 +180,25 @@ Referenced by:
 
 '''
 
-== parameter
```
[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply docs/_docs/sql-reference/grammar-reference.adoc:183]: yep, forgot to revert it. Thanks for the attention!

**Review by korlov42 — APPROVED** [review]
